tests/data/test_delta_dataset_source.py

```diff
@@ -113,3 +115,108 @@ def test_delta_dataset_source_too_many_inputs(spark_session, tmp_path):
 
     with pytest.raises(MlflowException, match='Must specify exactly one of "path" or "table_name"'):
         DeltaDatasetSource(path=tmp_path, delta_table_name="temp_delta_too_many_inputs")
+
+
+def test_uc_table_id_retrieval_works(spark_session, tmp_path):
+    def mock_resolve_table_name(table_name):
+        if table_name == "temp_delta_versioned_with_id":
+            return "default.temp_delta_versioned_with_id"
+        return table_name
+
+    def mock_lookup_table_id(table_name):
+        if table_name == "default.temp_delta_versioned_with_id":
+            return "uc_table_id_1"
+        return None
+
+    def mock_is_databricks_uc_table():
```

Comment:
Similarly can we mock something other than the full method here? Maybe we just need to mock `is_in_databricks_runtime` & force that to `True`

Reply:
Or maybe the problem is we can't properly run this logic since open source delta doesn't support three-level table names?

Reply:
Let me see what I can do.  It seems like a lot of work to validate a straightforward check.  Additionally, you would still need to mock the table name resolve method, so what exactly are you testing here?  Set membership of the elements in DATABRICKS_LOCAL_METASTORE_NAMES and DATABRICKS_SAMPLES_CATALOG_NAME?